src/kvstore/kvstore_ngraph.h

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+* Copyright 2018 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#ifndef MXNET_KVSTORE_KVSTORE_NGRAPH_H_
+#define MXNET_KVSTORE_KVSTORE_NGRAPH_H_
+
+#if MXNET_USE_NGRAPH_DISTRIBUTED
+
+#include <mxnet/kvstore.h>
+#include <mpi.h>
+
+namespace mxnet {
+namespace kvstore {
+
+/**
+ * \brief store data in local machine using nGraph
+ */
+class KVStoreNGRAPH : public KVStoreLocal {
```

Comment:
I wonder if it should be base on `public KVStore`. does Local kvstore only support single machine, which is what nccl kvstore based on? Dist sync allreduce uses `KVStore` (MX DO PR).

Reply:
So far as I understand, I think it doesn't matter since the real `push`/allreduce happens in the emitter. So I would change this to `public KVStore` for clarity.

Reply:
Our implementation is based on local kvstore. When a model is run with `mpirun`, each process gets its own kvstore. The communication takes place at the nGraph level. The other kvstore approaches are integrated with communication libraries directly.